src/Worker.Extensions.DurableTask/DefaultDurableClientContext.cs

```diff
@@ -91,30 +92,32 @@ public ValueTask<ConversionResult> ConvertAsync(ConverterContext converterContex
             // It's never expected to be wrong, but we code defensively just in case.
             if (converterContext.Source is not string clientConfigText)
             {
-                return new ValueTask<ConversionResult>(ConversionResult.Failed(new InvalidOperationException($"Expected the Durable Task WebJobs SDK extension to send a string payload for {nameof(DurableClientAttribute)}.")));
-            }
-
-            DurableClientInputData? inputData = JsonSerializer.Deserialize<DurableClientInputData>(clientConfigText);
-            if (string.IsNullOrEmpty(inputData?.rpcBaseUrl))
-            {
-                InvalidOperationException exception = new("Failed to parse the input binding payload data");
-                return new ValueTask<ConversionResult>(ConversionResult.Failed(exception));
+                return new ValueTask<ConversionResult>(ConversionResult.Failed(
+                    new InvalidOperationException($"Expected the Durable Task WebJobs SDK extension to send a string payload for {nameof(DurableClientAttribute)}.")));
             }
 
             try
             {
-                DurableTaskClient client = this.clientProvider.GetClient(inputData?.rpcBaseUrl);
+                DurableClientInputData? inputData = JsonSerializer.Deserialize<DurableClientInputData>(clientConfigText);
+                if (!Uri.TryCreate(inputData?.rpcBaseUrl, UriKind.Absolute, out Uri? endpoint))
+                {
+                    throw new InvalidOperationException("Failed to parse the input binding payload data");
```

Comment:
Why are you throwing here when in other places we return the exception as a `ConversionResult`?

Reply:
We will catch and return as `ConversionResult`. I can switch to just returning `ConversionResult`.